homeassistant/components/dhcp/__init__.py

```diff
@@ -143,7 +149,12 @@ async def _async_stop(*_):
 class WatcherBase:
     """Base class for dhcp and device tracker watching."""
 
-    def __init__(self, hass, address_data, integration_matchers):
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        address_data: dict[str, dict[str, str]],
+        integration_matchers: list[dict[str, str | bool]],
```

Comment:
We should use a TypedDict or similar to type the matchers.

Reply:
done